server/coordinator.go

```diff
@@ -446,3 +455,80 @@ func collectOperatorCounterMetrics(op Operator) {
 		operatorCounter.WithLabelValues(op.GetName(), op.GetState().String()).Add(1)
 	}
 }
+
+type heartbeatStream interface {
+	Send(*pdpb.RegionHeartbeatResponse) error
+}
+
+type streamUpdate struct {
+	storeID uint64
+	stream  heartbeatStream
+}
+
+type heartbeatStreams struct {
+	ctx       context.Context
+	clusterID uint64
+	streams   map[uint64]heartbeatStream
+	msgCh     chan *pdpb.RegionHeartbeatResponse
+	streamCh  chan streamUpdate
+}
+
+func newHeartbeatStreams(ctx context.Context, clusterID uint64) *heartbeatStreams {
+	hs := &heartbeatStreams{
+		ctx:       ctx,
+		clusterID: clusterID,
+		streams:   make(map[uint64]heartbeatStream),
+		msgCh:     make(chan *pdpb.RegionHeartbeatResponse, regionheartbeatSendChanCap),
+		streamCh:  make(chan streamUpdate, 1),
+	}
+	go hs.run()
+	return hs
+}
+
+func (s *heartbeatStreams) run() {
+	for {
+		select {
+		case update := <-s.streamCh:
+			s.streams[update.storeID] = update.stream
```

Comment:
No need to. `ServerStream` does not have `Close` function, it will close automatically after eof.